src/bsp/src/stm32/src/interloc/src/UWBRxFrame.cpp

```diff
@@ -13,4 +13,33 @@ float UWBRxFrame::getAccumulatorAngle() const {
 
     // TODO: Maybe change for LUT
     return atan2f((float)accumulatorI, (float(accumulatorQ)));
-}
\ No newline at end of file
+}
+
+float UWBRxFrame::getLOSConfidence() const {
+    // See DW1000 user manual p.45-46
+    // https://www.decawave.com/sites/default/files/resources/dw1000_user_manual_2.11.pdf
+    constexpr float A = 121.74; // We are using a 64Mhz PRF
+    constexpr float Z = 131072; // 2^17
+
+    float f1_2 = (float)m_fpAmpl1 * (float)m_fpAmpl1;
+    float f2_2 = (float)m_fpAmpl2 * (float)m_fpAmpl2;
+    float f3_2 = (float)m_fpAmpl3 * (float)m_fpAmpl3;
+    float n_2 = (float)m_rxPreambleCount * (float)m_rxPreambleCount;
```

Comment:
It doesn't cast it implicitly since you are assigning it to a float?

Reply:
It would cast it but only after the operation. So because we are multiplying two uint16_t together, if the result is bigger then 65000, it would not work